src/main/java/com/icthh/xm/tmf/ms/activation/service/SagaServiceImpl.java

```diff
@@ -87,14 +107,33 @@ public void onSagaEvent(SagaEvent sagaEvent) {
         List<BiFunction<SagaEvent, Context, Boolean>> preconditions = asList(this::isTransactionExists,
                                                                              this::isTransactionInCorrectState,
                                                                              this::isCurrentTaskNotFinished,
-                                                                             this::isAllDependsTaskFinished);
+                                                                             this::isAllDependsTaskFinished,
+                                                                             this::isTaskNotSuspended);
 
-        for (val precondition : preconditions) {
+        for (var precondition : preconditions) {
             if (!precondition.apply(sagaEvent, context)) {
                 return;
             }
         }
 
+        /**
+         This check avoid duplication executing event.
+         As example after resend events marked as in queue {@link #resendAllEventsInQueue()}
+         This check valid in multi node environment, because transaction id used as partition key in kafka.
+         Case with cluster rebalance known and acceptable.
+         */
+        if (TRUE.equals(executingTask.putIfAbsent(sagaEvent.getId(), true))) {
+            log.warn("Message {} already executing", sagaEvent);
```

Comment:
suggest ton split warning message with the content: 
            log.warn("Message already executing: {}", sagaEvent);